src/bun.js/bindings/bindings.cpp

```diff
@@ -830,6 +830,24 @@ bool Bun__deepEquals(JSC__JSGlobalObject* globalObject, JSValue v1, JSValue v2,
 
         return false;
     }
+    case ErrorInstanceType: {
+        if (c2Type != ErrorInstanceType) {
+            return false;
+        }
+
+        if (JSC::ErrorInstance* left = jsDynamicCast<JSC::ErrorInstance*>(v1)) {
+            JSC::ErrorInstance* right = jsDynamicCast<JSC::ErrorInstance*>(v2);
+
+            if (UNLIKELY(!right)) {
+                return false;
+            }
+
+            return (
+                left->sanitizedNameString(globalObject) == right->sanitizedNameString(globalObject) &&
+                left->sanitizedMessageString(globalObject) == right->sanitizedMessageString(globalObject)
+            );
```

Comment:
This is based on Node's implementation:

https://github.com/nodejs/node/blob/5e6448310d228aab139d1a5cb62ef5b771ca2c4b/lib/internal/util/comparisons.js#L235-L242